main.tf

```
@@ -26,7 +26,7 @@ resource "aws_s3_bucket_server_side_encryption_configuration" "athena-workspace"
 }
 
 resource "aws_s3_bucket_lifecycle_configuration" "athena-workspace" {
-  for_each = var.workspace_bucket_expiration_days != null ? [var.workspace_bucket_expiration_days] : []
+  for_each = toset(var.workspace_bucket_expiration_days != null ? [tostring(var.workspace_bucket_expiration_days)] : [])
```

in the end, that's just an optional resource: the typical pattern for this would be using `count`:

```suggestion
  count = var.workspace_bucket_expiration_days != null ? 1 : 0
```

and then use `var.workspace_bucket_expiration_days` directly for `rule.id`

---

done in 007b104ffb6550a0aba9b36c91a1cba68